src/internal/async-map-parallel.mjs

```diff
@@ -0,0 +1,80 @@
+import CircularBuffer from './circular-buffer'
+import asyncify from './asyncify'
+
+class ParallelMapper {
+  constructor (iterable, mapFn, concurrency) {
+    this.iter = asyncify(iterable)[Symbol.asyncIterator]()
+    this.buffer = new CircularBuffer(concurrency)
+    this.mapFn = mapFn
+    this.done = false
+    this.previousUnresolvedPromise = null
+  }
+
+  next () {
+    const { iter, buffer, done, mapFn } = this
+    let promise
+
+    if (buffer.isFull() || done) {
+      promise = buffer.shift()
+
+      if (!done) {
+        // Keep the buffer full!
+        buffer.push(this.next())
+      }
+    } else {
+      promise = iter.next()
+        .then(async item => {
+          // We are guarnteed by the async generator impl that this then block
+          // will be executed in order
+          const { previousUnresolvedPromise } = this
+
+          this.previousUnresolvedPromise = promise
+
+          let mapPromise
+          if (item.done) {
+            this.done = true
+          } else {
+            mapPromise = mapFn(item.value)
+
+            if (!buffer.isFull()) {
+              // Go on filling up the buffer in the background
+              // Note that we don't await the last generator item before requesting the next one.
+              // This makes the code "eager". It works because async generators queue `next` calls
+              // internally, as if they were requests. If there is a request in the queue, they
+              // begin to generate the next item as soon as the last one is finished, whether or not
+              // anybody has tried to consume it from ParallelMapper yet. This is what gives us
+              // us parallelism. Without the ability to read ahead in the source, we would have
+              // to wait until we had produced one mapped item until our consumer could call our
+              // next method again triggering another source fetch. This would result in completely
+              // serial behavior.
+              buffer.push(this.next())
+            }
+          }
+
+          let mappedValue
+          if (!item.done) {
+            mappedValue = await mapPromise
```

Comment:
you are awaiting the return of of the mapFn. This was not how it was working before.
The point of asyncMap - asyncFilter was to running the these functions in parallel

Reply:
Never mind, I think I understand now. Giving that you are inside a "then" callback, this will happen in parallel